http-client/src/transport.rs

```diff
@@ -167,4 +156,17 @@ mod tests {
 		let response = client.send_request(request).await.unwrap_err();
 		assert!(matches!(response, Error::RequestTooLarge));
 	}
+
+	#[tokio::test]
```

Comment:
maybe a little bit heavy for a test

Reply:
I can remove it but just to demonstrate that it works :)

Reply:
I'd move it to the separate `tests`, it's more of integration test than unit test.